src/hotspot/share/opto/regmask.hpp

```diff
@@ -128,54 +272,50 @@ class RegMask {
     _hwm = _RM_MAX;
     while (_hwm > 0      && _RM_UP[_hwm] == 0) _hwm--;
     while ((_lwm < _hwm) && _RM_UP[_lwm] == 0) _lwm++;
-    assert(valid_watermarks(), "post-condition");
-  }
-
-  // Handy copying constructor
-  RegMask(RegMask *rm) {
-    _hwm = rm->_hwm;
-    _lwm = rm->_lwm;
-    for (unsigned i = 0; i < _RM_SIZE; i++) {
-      _RM_UP[i] = rm->_RM_UP[i];
-    }
+    // For historical reasons, this constructor uses the last bit of the mask
+    // itself as the _all_stack flag. We need to record this fact using the now
+    // separate _all_stack flag.
+    set_AllStack(_RM_UP[_RM_MAX] & (uintptr_t(1) << _WordBitMask));
```

Comment:
As we discussed offline, I suggest simplifying this by updating the ADLC code as well to set `_all_stack` explicitly. Here is a patch that accomplishes that: https://github.com/robcasloz/jdk/commit/835628893ed5ea838c16afa49adbd57e4152a5dd, feel free to merge if you agree. The patch passes tier1-3 for all Oracle-supported platforms.

Reply:
Yes, nice cleanup. Now included!